tests/test_tuple.cpp

```diff
@@ -102,3 +102,68 @@ TEST_CASE("count_edge_on_boundary", "[test_tuple]")
     REQUIRE(edges.size() == 10);
     REQUIRE(cnt == 9);
 }
+
+TEST_CASE("tuple_iterator", "[test_tuple]")
+{
+    class IterableMesh : public TetMesh
+    {
+    public:
+        struct EdgeIterator
```

Comment:
I find it easier to write a foreach_edges function with a custom callback function rather than have to implement your own iterator type.

Reply:
In fact, this PR is meant to initiate a discussion on how we want to proceed with the interface. That's why I am prototyping in a testing section.